File: HardwareVault_FrontPage/src/types/device.ts
// types/device.ts
import { Manufacturer } from './manufacturer';

export interface Device {
  deviceId: string;
  ramSizeInMB: number;
  storageSizeInGB: number;
  storageType: 'SSD' | 'HDD';
  weightInKg: number;
  createdAt: string;
  updatedAt: string;
  cpu: {
    cpuId: number;
    modelName: string;
    manufacturer: Manufacturer;
  };
  gpu: {
    gpuId: number;
    modelName: string;
    manufacturer: Manufacturer;
  };
  powerSupply: {
    powerSupplyId: number;
    wattageInWatts: number;
  };
  usbPorts: UsbPort[];
}

export interface UsbPort {
  deviceUsbPortId: string;
  usbPortType: string;
  count: number;
}

export interface DeviceFilters {
  cpuManufacturer?: string;
  gpuManufacturer?: string;
  storageType?: 'SSD' | 'HDD';
  minRamInGB?: number;
  search?: string;
  page: number;
  pageSize: number;
}

export interface CreateDeviceDto {
  ramSizeInMB: number;
  storageSizeInGB: number;
  storageType: 'SSD' | 'HDD';
  weightInKg: number;
  cpuId: number;
  gpuId: number;
  powerSupplyWattageInWatts: number;
  usbPorts: CreateUsbPortDto[];
}

export interface CreateUsbPortDto {
  usbPortType: string;
  count: number;
}

export interface UpdateDeviceDto {
  ramSizeInMB?: number;
  storageSizeInGB?: number;
  storageType?: 'SSD' | 'HDD';
  weightInKg?: number;
  cpuId?: number;
  gpuId?: number;
  powerSupplyWattageInWatts?: number;
  usbPorts?: CreateUsbPortDto[];
}

export interface DeviceStatistics {
  totalDevices: number;
  activeDevices: number;
  recentImportJobsCount: number;
  lastImportDate?: string;
  devicesByCpuManufacturer: Record<string, number>;
  devicesByStorageType: Record<string, number>;
}

File: HardwareVault_Services/Domain/Entities/Device.cs
namespace HardwareVault_Services.Domain.Entities
{
    public class Device
    {
        public Guid Id { get; set; }
        public int RamSizeInMB { get; set; }
        public int StorageSizeInGB { get; set; }
        public string StorageType { get; set; } = string.Empty;
        public int CpuId { get; set; }
        public int GpuId { get; set; }
        public int PowerSupplyId { get; set; }
        public decimal WeightInKg { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public bool IsDeleted { get; set; }

        // Navigation properties
        public Cpu Cpu { get; set; } = null!;
        public Gpu Gpu { get; set; } = null!;
        public PowerSupply PowerSupply { get; set; } = null!;
        public ICollection<DeviceUsbPort> DeviceUsbPorts { get; set; } = new List<DeviceUsbPort>();
    }
}
